### DISCUSSION TITLE
DS18B20 no longer reporting after upgrading to version 14.2.0

### DISCUSSION THREAD

**Discussion by VE5BNC** [arendst/Tasmota#22156] [Q&A]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I'm using a Topgreener smart plug flashed with Tasmota with a DS18B20 temperature sensor on GPIO2.  I'm using the pre-compiled binaries.  I've been using the upgrade feature provided by the device GUI using the OTA option for the upgrade from 14.1 to 14.2 and a locally download copy of the bin files to downgrade from 14.2 to 14.1.  

When I upgraded from version 14.1.0 the DS18B20 no longer appears on the GUI and is not reported to my MQTT server.  I downgraded from version 14.2.0 to 14.1.0 and the DS18B20 is displaying and reporting correctly again.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): TopGreener TGWF115APM Power Monitoring Plug
- [X] Tasmota binary firmware version number used: 14.2.0
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: GUI via OTS and locally downloaded files
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
15:38:11.655 MQT: stat/GARAGE/RESULT = {"NAME":"TGWF115APM","GPIO":[0,320,1312,32,2720,2656,0,0,2624,321,224,0,0,0],"FLAG":0,"BASE":18}
15:38:11.868 MQT: stat/GARAGE/RESULT = {"Module":{"0":"TGWF115APM"}}
15:38:12.125 MQT: stat/GARAGE/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"320":"Led_i1"},"GPIO2":{"1312":"DS18x201"},"GPIO3":{"32":"Button1"},"GPIO4":{"2720":"BL0937 CF"},"GPIO5":{"2656":"HLWBL CF1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"2624":"HLWBL SEL_i"},"GPIO13":{"321":"Led_i2"},"GPIO14":{"224":"Relay1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}



```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
15:40:11.478 MQT: stat/GARAGE/STATUS = {"Status":{"Module":0,"DeviceName":"Garage","FriendlyName":["Garage"],"Topic":"GARAGE","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
15:40:11.515 MQT: stat/GARAGE/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:03:11","StartupUTC":"2024-09-16T21:37:00","Sleep":50,"CfgHolder":4617,"BootCount":90,"BCResetTime":"2021-02-18T03:31:01","SaveCount":1128,"SaveAddress":"FA000"}}
15:40:11.549 MQT: stat/GARAGE/STATUS2 = {"StatusFWR":{"Version":"14.2.0(release-tasmota)","BuildDateTime":"2024-08-14T12:36:35","Boot":31,"Core":"2_7_7","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"347/699"}}
15:40:11.569 MQT: stat/GARAGE/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["LP",""],"TelePeriod":60,"Resolution":"558180C0","SetOption":["00208009","2805C80001000600003C5A0A002800000000","00008281","00006000","08004000","00000000"]}}
15:40:11.605 MQT: stat/GARAGE/STATUS4 = {"StatusMEM":{"ProgramSize":648,"Free":352,"Heap":23,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,3,!4,!5,!6,7,!8,9,10,12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
15:40:11.643 MQT: stat/GARAGE/STATUS5 = {"StatusNET":{"Hostname":"GARAGE","IPAddress":"172.16.1.2","Gateway":"172.16.1.254","Subnetmask":"255.255.255.0","DNSServer1":"172.16.1.250","DNSServer2":"0.0.0.0","Mac":"84:0D:8E:4C:92:2F","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
15:40:11.666 MQT: stat/GARAGE/STATUS6 = {"StatusMQT":{"MqttHost":"172.16.1.215","MqttPort":1883,"MqttClientMask":"Garage","MqttClient":"Garage","MqttUser":"LPMQTT","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
15:40:11.691 MQT: stat/GARAGE/STATUS7 = {"StatusTIM":{"UTC":"2024-09-16T21:40:11Z","Local":"2024-09-16T15:40:11","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"-06:00","Sunrise":"06:43","Sunset":"19:17"}}
15:40:11.707 MQT: stat/GARAGE/STATUS9 = {"StatusPTH":{"PowerDelta":0,"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0,"MaxPower":0,"MaxPowerHold":10,"MaxPowerWindow":30,"MaxEnergy":0,"MaxEnergyStart":0}}
15:40:11.734 MQT: stat/GARAGE/STATUS10 = {"StatusSNS":{"Time":"2024-09-16T15:40:11","ENERGY":{"TotalStartTime":"2024-05-01T10:32:05","Total":106.937,"Yesterday":0.000,"Today":0.000,"Period":0,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":119,"Current":0.000}}}
15:40:11.756 MQT: stat/GARAGE/STATUS11 = {"StatusSTS":{"Time":"2024-09-16T15:40:11","Uptime":"0T00:03:11","UptimeSec":191,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"LP","BSSId":"C0:3F:0E:DA:D0:1A","Channel":11,"Mode":"11n","RSSI":52,"Signal":-74,"LinkCount":1,"Downtime":"0T00:00:05"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Upgrade from version 14.1.0 to 14.2.0.  Downgrading from version 14.2.0 to 14.1.0 restores the function of the DS18B20.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

![Garage - Main Menu 14 1 0](https://github.com/user-attachments/assets/f11f1a47-cc2e-4f51-b805-fc329f80ff8c)
![Garage - Main Menu 14 2 0](https://github.com/user-attachments/assets/b7db9826-0d29-407c-93d9-e025f8071a01)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by reimerp** [comment]
No idea, at least its compiiled in Tasmota 14.2.0 (release-tasmota) on ESP8266EX (ESP-01 18x20) Board. Mine is working 
`{"Time":"2024-09-26T21:42:06","DS18B20":{"Id":"000F98431670","Temperature":20.9}`
